cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGenerator.java

```diff
@@ -0,0 +1,220 @@
+  static final String ADDRESS_DEFAULT = "127.0.0.1";
+  static final int PORT_DEFAULT = 11030;
+  // HttpHandler has only one caller, and is accessed infrequently so one worker thread is sufficient
+  static final int WORKER_POOL_SIZE_DEFAULT = 1;
+  static final int EXEC_POOL_SIZE_DEFAULT = 0;
+  private NettyHttpService httpService;
+  private Codec<AccessToken> tokenCodec;
+  private TokenManager tokenManager;
+
+  private void doMain(String host, int port, int workerThreadPoolSize, int execThreadPoolSize) throws Exception {
+    CompletableFuture<?> completion = new CompletableFuture<>();
+    init(host, port, workerThreadPoolSize, execThreadPoolSize, completion);
+    httpService.start();
+    // Wait forever
+    completion.wait();
+    httpService.stop();
+  }
+
+  @VisibleForTesting
+  void init(String host, int port, int workerThreadPoolSize,
+            int execThreadPoolSize, CompletableFuture<?> completion) {
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(
+      new ConfigModule(),
+      new IOModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          // We don't do anything with metrics, so just bind to the no-op implementation
+          bind(MetricsCollectionService.class).to(NoOpMetricsCollectionService.class);
+        }
+      },
+      CoreSecurityRuntimeModule.getDistributedModule(cConf));
+    tokenManager = injector.getInstance(TokenManager.class);
+    tokenCodec = injector.getInstance(Key.get(new TypeLiteral<Codec<AccessToken>>() { }));
+
+    CommonNettyHttpServiceFactory factory = injector.getInstance(CommonNettyHttpServiceFactory.class);
+    NettyHttpService.Builder builder = factory.builder("accesstoken.generator")
+      .setHost(host)
+      .setPort(port)
+      .setExecThreadPoolSize(execThreadPoolSize)
+      .setWorkerThreadPoolSize(workerThreadPoolSize)
+      .setHttpHandlers(new TokenGeneratorHttpHandler(cConf, tokenManager, tokenCodec, completion));
+    if (cConf.getBoolean(Constants.Security.SSL.INTERNAL_ENABLED)) {
+      SConfiguration sConf = injector.getInstance(SConfiguration.class);
+      new HttpsEnabler().configureKeyStore(cConf, sConf).enable(builder);
+    }
+    httpService = builder.build();
+  }
+
+  @VisibleForTesting
+  NettyHttpService getService() {
+    return httpService;
+  }
+
+  @VisibleForTesting
+  TokenManager getTokenManager() {
+    return tokenManager;
+  }
+
+  @VisibleForTesting
+  Codec<AccessToken> getTokenCodec() {
+    return tokenCodec;
+  }
+
+  @Path(Constants.Gateway.INTERNAL_API_VERSION_3 + "/accesstoken")
+  public static class TokenGeneratorHttpHandler extends AbstractHttpHandler {
+    private final CConfiguration cConf;
+    private final TokenManager tokenManager;
+    private final Codec<AccessToken> tokenCodec;
+    private final CompletableFuture<?> completion;
+
+    public TokenGeneratorHttpHandler(CConfiguration cConf, TokenManager tokenManager,
```

Comment:
Seems like the fields can be injected instead of getting them from the Injector above and pass here?

Reply:
fixed